[2024-10-01T15:29:01.547+0000] {local_task_job_runner.py:123} INFO - ::group::Pre task execution logs
[2024-10-01T15:29:01.577+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: impression_scraper_dag.scraper_task manual__2024-10-01T15:28:55.503601+00:00 [queued]>
[2024-10-01T15:29:01.589+0000] {taskinstance.py:2612} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: impression_scraper_dag.scraper_task manual__2024-10-01T15:28:55.503601+00:00 [queued]>
[2024-10-01T15:29:01.589+0000] {taskinstance.py:2865} INFO - Starting attempt 1 of 1
[2024-10-01T15:29:01.614+0000] {taskinstance.py:2888} INFO - Executing <Task(PythonOperator): scraper_task> on 2024-10-01 15:28:55.503601+00:00
[2024-10-01T15:29:01.634+0000] {standard_task_runner.py:104} INFO - Running: ['***', 'tasks', 'run', 'impression_scraper_dag', 'scraper_task', 'manual__2024-10-01T15:28:55.503601+00:00', '--job-id', '6', '--raw', '--subdir', 'DAGS_FOLDER/impression.py', '--cfg-path', '/tmp/tmp_3mmcafk']
[2024-10-01T15:29:01.674+0000] {standard_task_runner.py:105} INFO - Job 6: Subtask scraper_task
[2024-10-01T15:29:01.679+0000] {logging_mixin.py:190} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:70 DeprecationWarning: This process (pid=234) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2024-10-01T15:29:01.682+0000] {standard_task_runner.py:72} INFO - Started process 240 to run task
[2024-10-01T15:29:02.071+0000] {task_command.py:467} INFO - Running <TaskInstance: impression_scraper_dag.scraper_task manual__2024-10-01T15:28:55.503601+00:00 [running]> on host 81290100fb3d
[2024-10-01T15:29:02.358+0000] {taskinstance.py:3131} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='impression_scraper_dag' AIRFLOW_CTX_TASK_ID='scraper_task' AIRFLOW_CTX_EXECUTION_DATE='2024-10-01T15:28:55.503601+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-10-01T15:28:55.503601+00:00'
[2024-10-01T15:29:02.360+0000] {taskinstance.py:731} INFO - ::endgroup::
[2024-10-01T15:29:03.279+0000] {_client.py:1026} INFO - HTTP Request: GET https://www.givingeurope.com/nl/nl/ "HTTP/1.1 200 OK"
[2024-10-01T15:29:03.321+0000] {logging_mixin.py:190} INFO - There are still 158 urls to be scraped.
[2024-10-01T15:29:03.488+0000] {_client.py:1026} INFO - HTTP Request: GET https://www.givingeurope.com/nl/nl/all-products/tassen-reizen/rugtassen/rugtassen?p=1&product_list_limit=48 "HTTP/1.1 200 OK"
[2024-10-01T15:29:03.536+0000] {logging_mixin.py:190} INFO - This is page number 1.
[2024-10-01T15:29:03.891+0000] {_client.py:1026} INFO - HTTP Request: GET https://components.givingeurope.com/api/v1/products/970950/configurator?locale=nl_NL&layout=wholesale "HTTP/1.1 200 OK"
[2024-10-01T15:29:03.895+0000] {taskinstance.py:3310} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 406, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/impression_scraper.py", line 227, in main
    [stock_detailed_info.append(s) for s in stock_gen_info]
                                            ^^^^^^^^^^^^^^
  File "/opt/airflow/dags/impression_scraper.py", line 87, in get_detailed_stock
    parsed_json = parse_api(url_api)
                  ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/tenacity/__init__.py", line 336, in wrapped_f
    return copy(f, *args, **kw)
           ^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/tenacity/__init__.py", line 475, in __call__
    do = self.iter(retry_state=retry_state)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/tenacity/__init__.py", line 376, in iter
    result = action(retry_state)
             ^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/tenacity/__init__.py", line 358, in _run_retry
    self.iter_state.retry_run_result = self.retry(retry_state)
                                       ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/tenacity/retry.py", line 272, in __call__
    return any(r(retry_state) for r in self.retries)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/tenacity/retry.py", line 272, in <genexpr>
    return any(r(retry_state) for r in self.retries)
               ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/tenacity/retry.py", line 183, in __call__
    return self.predicate(retry_state.outcome.result())
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/impression_scraper.py", line 26, in is_retryable_status_code
    return response.status_code in [500, 502, 503, 504, ]
           ^^^^^^^^^^^^^^^^^^^^
AttributeError: 'dict' object has no attribute 'status_code'
[2024-10-01T15:29:03.933+0000] {taskinstance.py:1225} INFO - Marking task as FAILED. dag_id=impression_scraper_dag, task_id=scraper_task, run_id=manual__2024-10-01T15:28:55.503601+00:00, execution_date=20241001T152855, start_date=20241001T152901, end_date=20241001T152903
[2024-10-01T15:29:03.951+0000] {taskinstance.py:340} INFO - ::group::Post task execution logs
[2024-10-01T15:29:03.951+0000] {standard_task_runner.py:124} ERROR - Failed to execute job 6 for task scraper_task ('dict' object has no attribute 'status_code'; 240)
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py", line 117, in _start_by_fork
    ret = args.func(args, dag=self.dag)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/cli_config.py", line 49, in command
    return func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/cli.py", line 115, in wrapper
    return f(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 483, in task_run
    task_return_code = _run_task_by_selected_method(args, _dag, ti)
                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 256, in _run_task_by_selected_method
    return _run_raw_task(args, ti)
           ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/cli/commands/task_command.py", line 341, in _run_raw_task
    return ti._run_raw_task(
           ^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/session.py", line 97, in wrapper
    return func(*args, session=session, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3004, in _run_raw_task
    return _run_raw_task(
           ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 273, in _run_raw_task
    TaskInstance._execute_task_with_callbacks(
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3158, in _execute_task_with_callbacks
    result = self._execute_task(context, task_orig)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 3182, in _execute_task
    return _execute_task(self, context, task_orig)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 767, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 733, in _execute_callable
    return ExecutionCallableRunner(
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 406, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 238, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 256, in execute_callable
    return runner.run(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/utils/operator_helpers.py", line 252, in run
    return self.func(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/impression_scraper.py", line 227, in main
    [stock_detailed_info.append(s) for s in stock_gen_info]
                                            ^^^^^^^^^^^^^^
  File "/opt/airflow/dags/impression_scraper.py", line 87, in get_detailed_stock
    parsed_json = parse_api(url_api)
                  ^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/tenacity/__init__.py", line 336, in wrapped_f
    return copy(f, *args, **kw)
           ^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/tenacity/__init__.py", line 475, in __call__
    do = self.iter(retry_state=retry_state)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/tenacity/__init__.py", line 376, in iter
    result = action(retry_state)
             ^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/tenacity/__init__.py", line 358, in _run_retry
    self.iter_state.retry_run_result = self.retry(retry_state)
                                       ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/tenacity/retry.py", line 272, in __call__
    return any(r(retry_state) for r in self.retries)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/tenacity/retry.py", line 272, in <genexpr>
    return any(r(retry_state) for r in self.retries)
               ^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/tenacity/retry.py", line 183, in __call__
    return self.predicate(retry_state.outcome.result())
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/impression_scraper.py", line 26, in is_retryable_status_code
    return response.status_code in [500, 502, 503, 504, ]
           ^^^^^^^^^^^^^^^^^^^^
AttributeError: 'dict' object has no attribute 'status_code'
[2024-10-01T15:29:03.993+0000] {local_task_job_runner.py:266} INFO - Task exited with return code 1
[2024-10-01T15:29:04.017+0000] {taskinstance.py:3900} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-10-01T15:29:04.018+0000] {local_task_job_runner.py:245} INFO - ::endgroup::
